### PR TITLE
Add mirror component

### PR DESCRIPTION
**Pull request by juunini**
# Doing

Mar-09-2022 09-55-34

- Add mirror AFRAME component
- playerHead visible and hide repeatedly
- In Reference, camera-tool.js line 605 and line 663

# Etc

- will be create spoke mirror component soon

# Example

```html
<a-plane mirror scale="5 5 5" position="0 3 0"></a-plane>
<a-circle mirror="color: #7f7f7f" scale="5 5 5" position="0 3 0"></a-circle>
```

### CODE REVIEW

File: src/components/mirror.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  schema: {
+    color: { type: "color", default: "#7f7f7f" }
+  },
+
+  init() {
+    const geometry = this.el.object3DMap?.mesh?.geometry;
+    if (!geometry) {
+      console.error('can not find "object3DMap.mesh.geometry"');
+      return;
+    }
```
Comment (hannut91) [inline src/components/mirror.js:49]: How about throwing an exception rather than simply printing the console?

File: src/components/mirror.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ */
+AFRAME.registerComponent("mirror", {
+  schema: {
+    color: { type: "color", default: "#7f7f7f" }
+  },
+
+  init() {
+    const geometry = this.el.object3DMap?.mesh?.geometry;
+    if (!geometry) {
+      console.error('can not find "object3DMap.mesh.geometry"');
```
Comment (hannut91) [inline src/components/mirror.js:47]: Strings in the code are using double quotes. How about using single quotes consistently?

```suggestion
      console.error("can not find 'object3DMap.mesh.geometry'");
```

Comment (juunini) [reply src/components/mirror.js:47]: image

Project's eslint rule T.T

File: src/components/mirror.js (hunk 1 above)
Comment (juunini) [reply src/components/mirror.js:49]: Good. I'll accept it

File: src/components/mirror.js (hunk 2 above)
Comment (juunini) [reply src/components/mirror.js]: Ah, I saw it wrong.
accept it. Thanks

**Comment by juunini** [comment]
# Add

## Doing

- [ADD] default geometry is `PlaneGeometry`
    - Currently, only `PlaneGeometry` is supported, but will support geometry primitives such as `circle` in the future.
- [ADD] `DEFAULT_TEXTURE_WIDTH` and `DEFAULT_TEXTURE_HEIGHT`

## Reason why set `DEFAULT_TEXTURE_WIDTH` and `DEFAULT_TEXTURE_HEIGHT`

|When set texture width and height|When not set(as default) texture width and height|
|-|-|
|스크린샷 2022-03-10 오후 6 28 45|스크린샷 2022-03-10 오후 6 29 08|
|mirror renders Anti-aliased|mirror renders aliased|

**Review by johnshaughnessy — APPROVED** [review]
lgtm! 

Do you have any thoughts for what we can do to prevent performance degrading on low-performance hardware like phones and tablets when one or more mirrors are in the scene?

**Comment by johnshaughnessy** [comment]
mirror

**Comment by johnshaughnessy** [comment]
I've edited the PR summary so that this feature is shown on the [what's new page](https://hubs.mozilla.com/whats-new)

**Comment by juunini** [comment]
- reduce view finder fps
- reduce texture width and height (aliased)
- when avatar is back of mirror, mirror not render. avatar is front of mirror, render mirror

**Pull request merged**